action/protocol/staking/protocol.go

```diff
@@ -458,7 +458,7 @@ func (p *Protocol) Validate(ctx context.Context, act action.Action, sr protocol.
 
 // ActiveCandidates returns all active candidates in candidate center
 func (p *Protocol) ActiveCandidates(ctx context.Context, sr protocol.StateReader, height uint64) (state.CandidateList, error) {
-	c, err := GetStakingStateReader(sr)
+	c, err := GetStakingStateReader(sr, true)
```

Comment:
For `StakingStateReader`, which sounds like a readonly component, why we need the extra parameter?

Reply:
in case the view does not exist, it will need to call `CreateBaseView`, which will call `NewCandidateCenter`, which requires the flag

Reply:
`GetStakingStateReader` calls `CreateBaseView`, which requires this flag

Reply:
`GetStakingStateReader ` has been removed in latest master